crates/noirc_evaluator/src/ssa/optimizations.rs

```diff
@@ -238,6 +238,16 @@ fn cse_block_with_anchor(
                         anchor.push_front(&ins.operation, *ins_id);
                     }
                 }
+                Operation::Result { .. } => {
+                    if let Some(similar) = anchor.find_similar_instruction(&operator) {
+                        debug_assert!(similar != ins.id);
```

Comment:
why is this a debug_assert instead of a regular assert?

Reply:
For no real reason, it could be a regular one as it is not supposed to happen. A debug_assert! has no impact so it is harmless and I supposed I added it during debugging. I will be happy to change it to a regular one if you prefer.

Reply:
A debug_assert gets removed on Release builds, so I was a bit confused :) I'd suggest we use assert! all over the codebase for things that shouldn't happen so they don't get optimised out, other than that happy to merge

Reply:
Done